cli/cli/Services/UnrealSourceGenerator/UnrealSourceGenerator.cs

```diff
@@ -936,13 +936,15 @@ private static void BuildSchemaIdCollisionMap(UEGenerationContext context)
 		context.SchemaNameCollisions.EnsureCapacity(context.Schemas.Count);
 		foreach (var namedOpenApiSchema in context.Schemas)
 		{
-			var name = namedOpenApiSchema.Schema.Extensions.TryGetValue(MICROSERVICE_EXTENSION_BEAMABLE_TYPE_NAME, out var nameOverride)
-				? new OpenApiReferenceId((nameOverride as OpenApiString).Value)
-				: namedOpenApiSchema.ReferenceId;
+			var name = namedOpenApiSchema.ReferenceId;
+			if (namedOpenApiSchema.Schema.Extensions.TryGetValue(MICROSERVICE_EXTENSION_BEAMABLE_TYPE_NAME, out var nameOverride))
+			{
+				name = new OpenApiReferenceId((nameOverride as OpenApiString).Value);
+			}
 
 			if (!context.SchemaNameCollisions.TryGetValue(name, out var list))
 			{
-				list = new List<NamedOpenApiSchema>(1);
+				list = new List<NamedOpenApiSchema>(8);

```

Comment:
lolwat?

Reply:
It pre-allocs a bit more memory to avoid stopping for re-allocate as we're building this list of name collisions (not that it matters that much compared to the fact that each `NamedOpenApiSchema` is individually heap-allocated). I was just reading through the function and noticed that.